src/mscorlib/src/System/String.cs

```diff
@@ -552,10 +549,13 @@ private unsafe static int CompareOrdinalHelper(String strA, String strB)
                 DiffOffset8: a += 4; b += 4;
                 DiffOffset4: a += 4; b += 4;
 #else
-                DiffOffset8: a += 2; b += 2;
-                DiffOffset6: a += 2; b += 2;
-                DiffOffset4: a += 2; b += 2;
-                DiffOffset2: a += 2; b += 2;
+                // Use jumps instead of falling through, since
+                // otherwise going to DiffOffset8 will involve
+                // 8 add instructions before getting to DiffNextInt
+                DiffOffset8: a += 8; b += 8; goto DiffOffset0;
+                DiffOffset6: a += 6; b += 6; goto DiffOffset0;
+                DiffOffset4: a += 4; b += 4; goto DiffOffset0;
```

Comment:
Were you able to measure whether the goto is beneficial for the `DiffOffset4` case? If it is beneficial for the `DiffOffset4` case, it should be beneficial for the `DiffOffset8` case in the BIT64 path as well.